src/App/docutils/ComponentPreview/EditableComponentPreview.js

```diff
@@ -0,0 +1,275 @@
+import React, { useState, useEffect } from "react";
+import PropTypes from "prop-types";
+import {
+    SandpackProvider,
+    SandpackCodeEditor,
+    SandpackPreview,
+} from "@codesandbox/sandpack-react";
+import { ShowCasePanelAdvanced } from "./components/ShowCasePanelAdvanced";
+import { useCodeParser } from "./hooksUtils/useCodeParser";
+import { resetOptions } from "./hooksUtils/optionsUtils";
+import { useJsSandboxInitCode } from "./hooksUtils/useSandboxInitCode";
+import packageJson from "~/package";
+
+// TODO: should the preview of showCasePanel & the one of advancedShowCasePanel use the same component ?
+// in a way it is using the same SndpackPreview, so it would keep them in sync (e.g. add additional button "copy to clipboard", etc)
+// on the other hand the whole container around is different (for some reason)
+const ShowCasePanel = ({
+    showCasePanelSm,
+    negative,
+    childrenPassed,
+    staticPreview,
+    previewMinHeight,
+}) => (
+    <div
+        className={`showcase-panel${showCasePanelSm ? " showcasepanel-sm" : ""}${
+            negative ? " showcase-panel-negative" : ""
+        }`}
+        style={
```

Comment:
Could there be a better way to set min-height? I have no solution right now, but it seems a bit "harsh" to pass in px height directly to inline class

Reply:
modifying CSS custom properties via inline style attribute is a pretty common practice (if I'm not mistaking), since the css custom properties (css variables) take priority via scoping proximity (instead of the usual CSS specificity)

this being said I _do_ feel what you're saying. Deep inside me it feels slightly "_dirty_" to modify style via inline style ... but so far I don't see any other way to fix it 🤔 any idea? if not, I think it is fine to keep it as is since it's a common css-variable pattern, and then iterating we might manage to make this nicer along the way?

Reply:
Alright! I just had to ask since we are almost never doing it overall the rest of DG, but no worries! Let's update it in the future IF we stumble upon something interesting :D